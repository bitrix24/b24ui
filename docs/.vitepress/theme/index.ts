/**
 * This is custom theme
 * We change some css
 * @see node_modules/vitepress/dist/client/theme-default/index.js
 */
import { ref, watch, toValue } from 'vue'
import { createRouter, createMemoryHistory } from 'vue-router'
import bitrix24UIPlugin from '@bitrix24/b24ui-nuxt/vue-plugin'
import Layout from './components/ui/Layout.vue'
import ComponentShowExample from './components/ui/ComponentShowExample.vue'
import ComponentProps from './components/ui/ComponentProps.vue'
import ComponentSlots from './components/ui/ComponentSlots.vue'
import ComponentEmits from './components/ui/ComponentEmits.vue'
import Description from './components/ui/Description.vue'
import Placeholder from './components/ui/Placeholder.vue'
import VPBadge from 'vitepress/dist/client/theme-default/components/VPBadge.vue'
import 'vitepress/dist/client/theme-default/styles/fonts.css'
import './tailwind.css'
// region ory theme ////
import './styles/components/custom-block.css'
import './styles/components/vp-code.css'
import './styles/components/vp-code-group.css'
import './styles/components/vp-doc.css'
import './styles/components/vp-sponsor.css'
// endregion ////
import './theme.css'

globalThis.useFetch = async (url: string, options: RequestInit & { params?: any, transform?: (data) => any } = {}) => {
  const data = ref()
  const status = ref('idle')
  async function _fetch() {
    status.value = 'loading'
    try {
      const params = options.params
        ? Object.fromEntries(
            Object.entries(options.params).map(([key, value]) => [key, toValue(value)])
          )
        : {}

      const queryString = new URLSearchParams(params as any).toString()
      const fullUrl = queryString ? `${url}?${queryString}` : url

      data.value = await fetch(fullUrl, options).then(r => r.json()).then(r => options.transform ? options.transform(r) : r)
      status.value = 'success'
    } catch (error) {
      console.error(error)
      status.value = 'error'
    }
  }
  _fetch()

  if (options?.params?.q) {
    watch(
      options.params.q,
      () => {
        _fetch()
      },
      { deep: true }
    )
  }

  return Promise.resolve({
    data,
    status
  })
}

export default {
  Layout,
  enhanceApp: async ({ app }) => {
    // region components ////
    app.component('Badge', VPBadge)
      .component('Description', Description)
      .component('ComponentShowExample', ComponentShowExample)
      .component('ComponentProps', ComponentProps)
      .component('ComponentSlots', ComponentSlots)
      .component('ComponentEmits', ComponentEmits)
      .component('Placeholder', Placeholder)
    // endregion ////

    if (!import.meta.env.SSR) {
      const router = createRouter({
        routes: [
          // { path: '', component: app }
        ],
        history: createMemoryHistory()
        // history: createWebHistory()
      })
      app.use(router)

      app.use(bitrix24UIPlugin)
    }
  }
}
