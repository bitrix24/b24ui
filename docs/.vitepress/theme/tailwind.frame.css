/*
@plugin '@bitrix24/b24style';
@plugin "@bitrix24/b24ui-nuxt";
*/
@import "tailwindcss" source("../../");
@import "@bitrix24/b24ui-nuxt";
/*
*/



