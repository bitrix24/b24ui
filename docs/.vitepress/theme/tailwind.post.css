@import "tailwindcss" source("../../../docs");
@import "@bitrix24/b24ui-nuxt";

/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}
