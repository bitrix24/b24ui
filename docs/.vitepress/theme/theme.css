/**
 * bxshef: Custom
 * -------------------------------------------------------------------------- */

:root
{
  --sh-scrollbar-thumb: #3c3c43;
  --sh-scrollbar-background: #ebebef;
}

.dark
{
  --vp-c-bg: #0f172a;
  --vp-c-bg-alt: #161618;
  --vp-c-bg-elv: #202127;
  --vp-c-bg-soft: #202127;
  --vp-code-block-bg: #1a2234;

  --sh-scrollbar-thumb: #bdc1c6;
  --sh-scrollbar-background: #3c3c43;
}

.bg-grid-example {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

.dark\:bg-grid-example:is(.dark *) {
  background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 32 32' width='32' height='32' fill='none' stroke='rgb(130 139 149 / 0.25)'%3e%3cpath d='M0 .5H31.5V32'/%3e%3c/svg%3e");
}

iframe
{
  min-width: 100%;
  border: none;
  scrollbar-width: thin;
  background-color: transparent;
}

/** region scrollbar */
* {
  scrollbar-color: var(--sh-scrollbar-thumb) var(--sh-scrollbar-background);
  scrollbar-width: thin;
}

/* Styling the scrollbar in Chrome and Safari */
*::-webkit-scrollbar {
  width: 6px;
}

/* Set thumb color for Chrome and Safari */
*::-webkit-scrollbar-thumb {
  background-color: var(--sh-scrollbar-thumb);
  border-radius: 4px;
}

/* Set track color for Chrome and Safari */
*::-webkit-scrollbar-track {
  background-color: var(--sh-scrollbar-background);
}
/** endregion */
