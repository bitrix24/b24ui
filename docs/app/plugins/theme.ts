export default defineNuxtPlugin({
  enforce: 'post',
  setup() {
  }
})
