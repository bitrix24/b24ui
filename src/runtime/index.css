@plugin '@bitrix24/b24style';
@import '#build/b24ui.css';
@import './keyframes.css';
@import './air-design-tokens/index.css';

@layer base {
  body,
  .sidebar-layout.--inner {
    /* @see air-design-tokens/???_b24_context*.css */
    --air-theme-background:
      var(--air-theme-bg-image, none)
      var(--air-theme-bg-position, 0 0) / var(--air-theme-bg-size, auto)
      var(--air-theme-bg-repeat, repeat)
      var(--air-theme-bg-attachment, scroll)
      var(--air-theme-bg-color, transparent)
    ;

    --air-theme-background-blurred:
      var(--air-theme-bg-image-blurred, none)
      var(--air-theme-bg-position, 0 0) / var(--air-theme-bg-size, auto)
      var(--air-theme-bg-repeat, repeat)
      var(--air-theme-bg-attachment, scroll)
    ;
  }

  /* @see src/theme/sidebar-layout.ts */
  body {
    background: var(--air-theme-background);
  }

  .sidebar-layout.--inner {
    background: var(--air-theme-background);
  }

  body {
    overflow-y: scroll!important;
    padding-right: 0!important;
    scrollbar-gutter: stable;
  }

  .--app[data-state="loading"] {
    overflow: hidden;
  }
}
