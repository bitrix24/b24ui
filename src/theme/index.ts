export { default as advice } from './advice'
export { default as avatar } from './avatar'
export { default as avatarGroup } from './avatar-group'
export { default as badge } from './badge'
export { default as button } from './button'
export { default as buttonGroup } from './button-group'
export { default as checkbox } from './checkbox'
export { default as chip } from './chip'
export { default as container } from './container'
export { default as countdown } from './countdown'
export { default as formField } from './form-field'
export { default as input } from './input'
export { default as kbd } from './kbd'
export { default as link } from './link'
export { default as progress } from './progress'
export { default as radioGroup } from './radio-group'
export { default as range } from './range'
export { default as separator } from './separator'
export { default as skeleton } from './skeleton'
export { default as switch } from './switch'
export { default as textarea } from './textarea'
export { default as tooltip } from './tooltip'
