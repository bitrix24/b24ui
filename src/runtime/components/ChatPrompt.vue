<script lang="ts">
import type { AppConfig } from '@nuxt/schema'
import theme from '#build/b24ui/chat-prompt'
import type { TextareaProps, TextareaSlots } from '../types'
import type { ComponentConfig } from '../types/tv'

type ChatPrompt = ComponentConfig<typeof theme, AppConfig, 'chatPrompt'>

export interface ChatPromptProps extends Pick<TextareaProps, 'rows' | 'autofocus' | 'autofocusDelay' | 'autoresize' | 'autoresizeDelay' | 'maxrows' | 'icon' | 'avatar' | 'loading' | 'disabled'> {
  /**
   * The element or component this component should render as.
   * @defaultValue 'form'
   */
  as?: any
  /**
   * The placeholder text for the textarea.
   * @defaultValue t('chatPrompt.placeholder')
   */
  placeholder?: string
  /**
   * @defaultValue 'outline'
   */
  variant?: ChatPrompt['variants']['variant']
  error?: Error
  class?: any
  b24ui?: ChatPrompt['slots'] & TextareaProps['b24ui']
}

export interface ChatPromptEmits {
  submit: [event: Event]
  close: [event: Event]
}

export interface ChatPromptSlots extends TextareaSlots {
  header(props?: {}): any
  footer(props?: {}): any
}
</script>

<script setup lang="ts">
import { computed, useTemplateRef } from 'vue'
import { Primitive, useForwardProps } from 'reka-ui'
import { reactivePick } from '@vueuse/core'
import { useAppConfig } from '#imports'
import { useLocale } from '../composables/useLocale'
import { omit, transformUI } from '../utils'
import { tv } from '../utils/tv'
import B24Textarea from './Textarea.vue'

defineOptions({ inheritAttrs: false })

const props = withDefaults(defineProps<ChatPromptProps>(), {
  as: 'form',
  autofocus: true,
  autoresize: true,
  rows: 1
})
const emits = defineEmits<ChatPromptEmits>()
const slots = defineSlots<ChatPromptSlots>()

const model = defineModel<string>({ default: '' })

const { t } = useLocale()
const appConfig = useAppConfig() as ChatPrompt['AppConfig']

const textareaProps = useForwardProps(reactivePick(props, 'rows', 'autofocus', 'autofocusDelay', 'autoresize', 'autoresizeDelay', 'maxrows', 'icon', 'avatar', 'loading'))

const getProxySlots = () => omit(slots, ['header', 'footer'])

const b24ui = computed(() => tv({ extend: tv(theme), ...(appConfig.b24ui?.chatPrompt || {}) })({
  variant: props.variant
}))

function submit(e: Event) {
  if (model.value.trim() === '') {
    return
  }

  emits('submit', e)
}

function blur(e: Event) {
  textarea.value?.textareaRef?.blur()

  emits('close', e)
}

const textarea = useTemplateRef('textarea')

defineExpose({
  textareaRef: textarea.value?.textareaRef
})
</script>

<template>
  <Primitive :as="as" :class="b24ui.root({ class: [props.b24ui?.root, props.class] })" @submit.prevent="submit">
    <div v-if="!!slots.header" :class="b24ui.header({ class: props.b24ui?.header })">
      <slot name="header" />
    </div>

    <B24Textarea
      ref="textarea"
      v-model="model"
      :placeholder="placeholder || t('chatPrompt.placeholder')"
      :disabled="Boolean(error) || disabled"
      no-padding
      no-border
      v-bind="{ ...textareaProps, ...$attrs }"
      :b24ui="transformUI(omit(b24ui, ['root', 'body', 'header', 'footer']), props.b24ui)"
      :class="b24ui.body({ class: props.b24ui?.body })"
      @keydown.enter.exact.prevent="submit"
      @keydown.esc="blur"
    >
      <template v-for="(_, name) in getProxySlots()" #[name]="slotData">
        <slot :name="name" v-bind="slotData" />
      </template>
    </B24Textarea>

    <div v-if="!!slots.footer" :class="b24ui.footer({ class: props.b24ui?.footer })">
      <slot name="footer" />
    </div>
  </Primitive>
</template>
