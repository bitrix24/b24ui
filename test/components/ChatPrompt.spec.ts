import { describe, it, expect } from 'vitest'
import { axe } from 'vitest-axe'
import { mountSuspended } from '@nuxt/test-utils/runtime'
import ChatPrompt from '../../src/runtime/components/ChatPrompt.vue'
import type { ChatPromptProps, ChatPromptSlots } from '../../src/runtime/components/ChatPrompt.vue'
import ComponentRender from '../component-render'
import theme from '#build/b24ui/chat-prompt'

describe('ChatPrompt', () => {
  const variants = Object.keys(theme.variants.variant) as any

  it.each([
    // Props
    ['with placeholder', { props: { placeholder: 'Placeholder' } }],
    ['with error', { props: { error: new Error('Error') } }],
    ...variants.map((variant: string) => [`with variant ${variant}`, { props: { variant } }]),
    ['with as', { props: { as: 'section' } }],
    ['with class', { props: { class: 'p-5' } }],
    ['with b24ui', { props: { b24ui: { footer: 'justify-end' } } }],
    // Slots
    ['with header slot', { slots: { header: () => 'Header slot' } }],
    ['with footer slot', { slots: { footer: () => 'Footer slot' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: ChatPromptProps, slots?: Partial<ChatPromptSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, ChatPrompt)
    expect(html).toMatchSnapshot()
  })

  it('passes accessibility tests', async () => {
    const wrapper = await mountSuspended(ChatPrompt, {
      props: {
        placeholder: 'Placeholder'
      }
    })

    expect(await axe(wrapper.element)).toHaveNoViolations()
  })
})
