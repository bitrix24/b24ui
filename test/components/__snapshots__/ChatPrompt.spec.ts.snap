// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChatPrompt > renders with as correctly 1`] = `
"<section class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-outline-bg) ring-(--ui-color-design-outline-stroke) ring-(length:--ui-design-outline-stroke-weight) text-(--ui-color-design-outline-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</section>"
`;

exports[`ChatPrompt > renders with b24ui correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-outline-bg) ring-(--ui-color-design-outline-stroke) ring-(length:--ui-design-outline-stroke-weight) text-(--ui-color-design-outline-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;

exports[`ChatPrompt > renders with class correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-outline-bg) ring-(--ui-color-design-outline-stroke) ring-(length:--ui-design-outline-stroke-weight) text-(--ui-color-design-outline-content) p-5">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;

exports[`ChatPrompt > renders with error correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-outline-bg) ring-(--ui-color-design-outline-stroke) ring-(length:--ui-design-outline-stroke-weight) text-(--ui-color-design-outline-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" disabled="" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;

exports[`ChatPrompt > renders with footer slot correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-outline-bg) ring-(--ui-color-design-outline-stroke) ring-(length:--ui-design-outline-stroke-weight) text-(--ui-color-design-outline-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <div class="flex items-center justify-between gap-1.5">Footer slot</div>
</form>"
`;

exports[`ChatPrompt > renders with header slot correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-outline-bg) ring-(--ui-color-design-outline-stroke) ring-(length:--ui-design-outline-stroke-weight) text-(--ui-color-design-outline-content)">
  <div class="flex items-center gap-1.5">Header slot</div>
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;

exports[`ChatPrompt > renders with placeholder correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-outline-bg) ring-(--ui-color-design-outline-stroke) ring-(length:--ui-design-outline-stroke-weight) text-(--ui-color-design-outline-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Placeholder" class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;

exports[`ChatPrompt > renders with variant filled correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-filled-na-bg) ring-(--ui-color-design-filled-na-stroke) ring-(length:--ui-design-filled-na-stroke-weight) text-(--ui-color-design-filled-na-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;

exports[`ChatPrompt > renders with variant outline correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-outline-bg) ring-(--ui-color-design-outline-stroke) ring-(length:--ui-design-outline-stroke-weight) text-(--ui-color-design-outline-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;

exports[`ChatPrompt > renders with variant plain correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-plain-bg) ring-(--ui-color-design-plain-stroke) ring-(length:--ui-design-plain-stroke-weight) text-(--ui-color-design-plain-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;

exports[`ChatPrompt > renders with variant tinted correctly 1`] = `
"<form class="relative flex flex-col items-stretch gap-2 px-2.5 py-2 w-full rounded-lg backdrop-blur bg-(--ui-color-design-tinted-na-bg) ring-(--ui-color-design-tinted-na-stroke) ring-(length:--ui-design-tinted-na-stroke-weight) text-(--ui-color-design-tinted-na-content)">
  <!--v-if-->
  <div class="isolate relative inline-flex items-start">
    <!--v-if--><textarea rows="1" placeholder="Enter your message here..." class="w-full pt-[7px] pb-[8px] border-0 focus:outline-none disabled:cursor-not-allowed disabled:pointer-events-none disabled:select-none disabled:opacity-30 disabled:resize-none appearance-none transition duration-300 ease-linear text-(--ui-color-base-1) placeholder:text-(--ui-color-design-plain-na-content-secondary) hover:text-(--ui-color-base-1) focus:text-(--ui-color-base-1) active:text-(--ui-color-base-1) font-[family-name:var(--ui-font-family-primary)] font-(--ui-font-weight-regular) align-middle group-not-only:not-first:after:content-[''] group-not-only:not-first:after:absolute group-not-only:not-first:after:top-[7px] group-not-only:not-first:after:bottom-[6px] group-not-only:not-first:after:left-0 group-not-only:not-first:after:w-px group-not-only:not-first:after:bg-current/30 resize-none style-filled rounded-(--ui-border-radius-sm) px-0 ring-0 focus-visible:ring-0 style-transparent-bg text-base/5" value=""></textarea>
    <!--v-if-->
    <!--v-if-->
  </div>
  <!--v-if-->
</form>"
`;
